picard/ui/scripteditor.py

```diff
@@ -0,0 +1,547 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+from functools import partial
+import os.path
+
+from PyQt5 import (
+    QtCore,
+    QtWidgets,
+)
+from PyQt5.QtGui import QPalette
+
+from picard import log
+from picard.config import (
+    TextOption,
+    get_config,
+)
+from picard.const import DEFAULT_FILE_NAMING_FORMAT
+from picard.file import File
+from picard.script import (
+    ScriptError,
+    ScriptParser,
+)
+from picard.util.settingsoverride import SettingsOverride
+
+from picard.ui import PicardDialog
+from picard.ui.options import OptionsPage
+from picard.ui.options.scripting import (
+    OptionsCheckError,
+    ScriptCheckError,
+)
+from picard.ui.ui_scripteditor import Ui_ScriptEditor
+from picard.ui.widgets.scriptdocumentation import ScriptingDocumentationWidget
+
+
+PRESET_SCRIPTS = [
+    {
+        "title": N_("Current file naming script saved in configuration"),
+        # Setting the script to None will force the current naming script from the configuration settings to be loaded.
+        "script": None
+    },
+    {
+        "title": N_("[Album artist]/[album]/[Track #]. [Title]"),
+        "script": """\
+%albumartist%/
+%album%/
+%tracknumber%. %title%"""
+    },
+    {
+        "title": N_("[Album]/[Disc and track #] [Artist] - [Title]"),
+        "script": """\
+$if2(%albumartist%,%artist%)/
+$if(%albumartist%,%album%/,)
+$if($gt(%totaldiscs%,1),%discnumber%-,)
+$if($and(%albumartist%,%tracknumber%),$num(%tracknumber%,2) ,)
+$if(%_multiartist%,%artist% - ,)
+%title%"""
+    },
+]
+
+
+class ScriptEditorExamples():
+    """File naming script examples.
+    """
+    def __init__(self, parent=None, tagger=None):
+        """File naming script examples.
+
+        Args:
+            parent (QMainWindow or OptionsPage, optional): Parent object. Defaults to None.
+            tagger (object, optional): Object containing the main window tagger object. Defaults to None.
+        """
+        self.tagger = tagger
+        self._sampled_example_files = []
+        config = get_config()
+        self.settings = config.setting
+        self.example_list = []
+
+    def update_sample_example_files(self):
+        """Get a new sample of randomly selected / loaded files to use as renaming examples.
+        """
+        import random
+        max_samples = 10  # pick up to 10 samples
```

Comment:
I would move this to class property and use it instead of hardcoded (10)  in following strings:

```
picard/ui/ui_options_renaming.py:        self.label_2.setText(_("If you select files from the Cluster pane or Album pane prior to opening the Options screen, up to 10 files will be randomly chosen from your selection as file naming examples.  If you have not selected any files, then some default examples will be provided."))
picard/ui/ui_scripteditor.py:        self.example_filename_sample_files_button.setToolTip(_("Up to 10 items chosen at random from files selected in the main window"))
```

Reply:
Set as a class property as suggested.  Also set the usage notes and tool tip strings as class properties so that they are consistent and only need to be translated once.